core/utils.ts

```diff
@@ -245,6 +245,110 @@ export function validateQueryString(session: Session, query: string, filename: s
   }
 }
 
+/**
+ * Checks if the Cloud Resource connection has a valid format.
+ * @param session Used to throw a compile error if connection is not valid.
+ * @param connection String to be validated.
+ * @param filename Used to throw a compile error if connection is not valid.
+ */
+export function validateConnectionFormat(
+  session: Session,
+  connection: string,
+  filename: string,
+) {
+  // Connection pattern of the form project.location.connection_id. Example:
+  // my-project.us-central1.my-connection
+  const dotPattern = /^[a-zA-Z0-9_-]+\.[a-zA-Z0-9_-]+\.[a-zA-Z0-9_-]+$/;
+
+  // Connection pattern of the form projects/<substring>/locations/<substring>/connections/<substring>
+  // Example: projects/my-project/locations/us-central1/connections/my-connection
+  // Substrings cannot contain '/'.
+  const resourcePattern =
+    /^projects\/[^/]+\/locations\/[^/]+\/connections\/[^/]+$/;
+
+  const isValidFormat =
+    dotPattern.test(connection) || resourcePattern.test(connection);
+
+  if (connection !== 'DEFAULT' && !isValidFormat) {
+    session.compileError(
+      new Error(
+        'The connection must be in the format `{project}.{location}.{connection_id}` or `projects/{project}/locations/{location}/connections/{connection_id}`, or be set to `DEFAULT`.',
+      ),
+      filename,
+    );
+  }
+
+}
+
+/**
+ * Checks if the storageUri is a valid GCS path.
+ * @param session Used to throw a compile error if storageUri is not valid.
+ * @param storageUri String to be validated.
+ * @param filename Used to throw a compile error if storageUri is not valid.
+ */
+export function validateStorageUriFormat(
+  session: Session,
+  storageUri: string,
+  filename: string,
+) {
+  // storageUri must have format gs://<bucket_name>/<path_to_data>
+  const gcsPathPattern = /^gs:\/\/([^/]+)\/(.+)$/;
+
+  if (!gcsPathPattern.test(storageUri)) {
+    session.compileError(
+      new Error(
+        'The storage URI must be in the format `gs://{bucket_name}/{path_to_data}`.',
+      ),
+      filename,
+    );
+  }
+}
+
+/**
+ * Returns a file format for an Iceberg table, as specified in the user's config file.
+ * @param configFileFormat User-provided file format, if it exists.
+ * @return File format used when creating an Iceberg table.
+ */
+export function getFileFormatValueForIcebergTable(
```

Comment:
I propose to have a logic here like: if FileFormat is defined and and not unknown, then use it, otherwise fallback to parquet.